gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/ExternalDataConfiguration.java

```diff
@@ -0,0 +1,419 @@
+ * By defining these properties, the data source can then be queried as if it were a standard
+ * BigQuery table. Support for external tables is experimental and might be subject to changes or
+ * removed.
+ *
+ * @see <a href="https://cloud.google.com/bigquery/federated-data-sources">Federated Data Sources
+ *     </a>
+ */
+public class ExternalDataConfiguration implements Serializable {
+
+  static final Function<com.google.api.services.bigquery.model.ExternalDataConfiguration,
+      ExternalDataConfiguration> FROM_PB_FUNCTION =
+      new Function<com.google.api.services.bigquery.model.ExternalDataConfiguration,
+          ExternalDataConfiguration>() {
+        @Override
+        public ExternalDataConfiguration apply(
+            com.google.api.services.bigquery.model.ExternalDataConfiguration configurationPb) {
+          return ExternalDataConfiguration.fromPb(configurationPb);
+        }
+      };
+  static final Function<ExternalDataConfiguration,
+      com.google.api.services.bigquery.model.ExternalDataConfiguration> TO_PB_FUNCTION =
+      new Function<ExternalDataConfiguration,
+          com.google.api.services.bigquery.model.ExternalDataConfiguration>() {
+        @Override
+        public com.google.api.services.bigquery.model.ExternalDataConfiguration apply(
+            ExternalDataConfiguration configuration) {
+          return configuration.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -8004288831035566549L;
+
+  private final List<String> sourceUris;
+  private final TableSchema schema;
+  private final String sourceFormat;
+  private final Integer maxBadRecords;
+  private final Boolean ignoreUnknownValues;
+  private final String compression;
+  private final CsvOptions csvOptions;
+
+  public static final class Builder {
+
+    private List<String> sourceUris;
+    private TableSchema schema;
+    private String sourceFormat;
+    private Integer maxBadRecords;
+    private Boolean ignoreUnknownValues;
+    private String compression;
+    private CsvOptions csvOptions;
+
+    private Builder() {}
+
+    /**
+     * Sets the fully-qualified URIs that point to your data in Google Cloud Storage. Each URI can
+     * contain one '*' wildcard character that must come after the bucket's name. Size limits
+     * related to load jobs apply to external data sources, plus an additional limit of 10 GB
+     * maximum size across all URIs.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/loading-data-into-bigquery#quota">Quota</a>
+     */
+    public Builder sourceUris(List<String> sourceUris) {
+      this.sourceUris = ImmutableList.copyOf(checkNotNull(sourceUris));
+      return this;
+    }
+
+    /**
+     * Sets the schema for the external data.
+     */
+    public Builder schema(TableSchema schema) {
+      this.schema = checkNotNull(schema);
+      return this;
+    }
+
+    /**
+     * Sets the source format of the external data.
```

Comment:
Yes I noticed that, and also an extract job supports [different destination formats](https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.extract.destinationFormat).